src/serializer/LazyObjectsSerializer.js

```diff
@@ -105,6 +118,121 @@ export class LazyObjectsSerializer extends ResidualHeapSerializer {
     return getOrDefault(this._valueLazyIds, obj, () => this._lazyObjectIdSeed++);
   }
 
+  _isLazyObject(val: Value) {
+    // Currently, only support lazy raw object.
+    return this._isRawObject(val);
+  }
+
+  // Must keep in-sync with ResidualHeapSerializer logic.
+  _isObject(val: Value) {
+    return !(
+      val instanceof AbstractValue ||
+      val.isIntrinsic() ||
+      val instanceof EmptyValue ||
+      val instanceof UndefinedValue ||
+      ResidualHeapInspector.isLeaf(val) ||
+      IsArray(this.realm, val) ||
+      val instanceof ProxyValue ||
+      val instanceof FunctionValue ||
+      val instanceof SymbolValue
+    );
+  }
+
+  // Must keep in-sync with ResidualHeapSerializer logic.
+  _isRawObject(val: Value) {
+    if (!this._isObject(val) || !(val instanceof ObjectValue) || val.originalConstructor !== undefined) {
+      return false;
+    }
+    invariant(val instanceof ObjectValue);
+
+    let kind = val.getKind();
+    switch (kind) {
+      case "RegExp":
+      case "Number":
+      case "String":
+      case "Boolean":
+      case "Date":
+      case "Float32Array":
+      case "Float64Array":
+      case "Int8Array":
+      case "Int16Array":
+      case "Int32Array":
+      case "Uint8Array":
+      case "Uint16Array":
+      case "Uint32Array":
+      case "Uint8ClampedArray":
+      case "DataView":
+      case "ArrayBuffer":
+      case "ReactElement":
+      case "Map":
+      case "WeakMap":
+      case "Set":
+      case "WeakSet":
+        return false;
```

Comment:
The giant switch seems unnecessary, as only `"Object"` or not matters.

Reply:
Makes sense. I will delete it.